### PR TITLE
Committing Food Feature

### PR DESCRIPTION
**Pull request by Muthunatesan**
Added New Food tile. 
Created a test to verify the tile is created 
Tested new tile color

### CODE REVIEW

File: src/main/java/engine/GameEngine.java [hunk 1]
```diff
@@ -17,6 +17,7 @@
 	private int levelHorizontalDimension;
 	private int levelVerticalDimension;
 	private Point player;
+	private Point food;
```
Comment (borisv13) [inline src/main/java/engine/GameEngine.java]: This tells me that there will only ever be ONE food, like there is only one player. Is this true?
If not, this should not be here.

File: src/main/java/engine/GameEngine.java [hunk 2]
```diff
@@ -36,6 +37,9 @@ public void addTile(int x, int y, TileType tileType) {
 		if (tileType.equals(TileType.PLAYER)) {
 			setPlayer(x, y);
 			tiles.put(new Point(x, y), TileType.PASSABLE);
+		} else if (tileType.equals(TileType.FOOD)) {
```
Comment (borisv13) [inline src/main/java/engine/GameEngine.java]: Same as the comment above.

File: src/main/java/ui/GamePanel.java [hunk 3]
```diff
@@ -36,6 +36,8 @@ public void paint(Graphics graphics) {
 		tilePainter.paintTiles(graphics, gameEngine, tileWidth, tileHeight);
 		tilePainter.paintPlayer(graphics, gameEngine.getPlayerXCoordinate(), gameEngine.getPlayerYCoordinate(),
 				tileWidth, tileHeight, TileType.PLAYER);
+		tilePainter.paintFood(graphics, gameEngine.getFoodXCoordinate(), gameEngine.getFoodYCoordinate(), tileWidth,
```
Comment (borisv13) [inline src/main/java/ui/GamePanel.java]: Same as the comment above. This should not be needed if you just added the TileType and TileColorMap.

File: src/test/resources/features/main/PlayerMovement.feature [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+      | XXXXXXXX |
+      | X      X |
+      | X     FX |
+      | X     PX |
+      | XXXXXXXX |
+
+
+  Scenario: Move up in the board
+    When the player moves up
+    Then the player is located at (6, 2) and F will be disappered
```
Comment (borisv13) [inline src/test/resources/features/main/PlayerMovement.feature]: "6,2" is not clear. Either make the map smaller to make it more clear, or change the wording.
AND should be on a separate line, with a separate assert.

File: src/main/java/engine/GameEngine.java (hunk 1 above)
Comment (Muthunatesan) [reply src/main/java/engine/GameEngine.java]: Yes, there will be ONE food tile at a time. There will not be two or more foods at same time.

File: src/main/java/engine/GameEngine.java (hunk 2 above)
Comment (Muthunatesan) [reply src/main/java/engine/GameEngine.java]: The same above applies

File: src/main/java/ui/GamePanel.java (hunk 3 above)
Comment (Muthunatesan) [reply src/main/java/ui/GamePanel.java]: If i try to remove it then food tile color will not visible and it leads to an error.

File: src/test/resources/features/main/PlayerMovement.feature (hunk 4 above)
Comment (Muthunatesan) [reply src/test/resources/features/main/PlayerMovement.feature]: Fixed - Reduced size and added an separate assert

**Comment by Muthunatesan** [comment]
Fixed the errors

**Comment by Muthunatesan** [comment]
Thanks. I have included both the features mentioned.
1. The player will be moved to next level when all the food tiles are collected.
2. The player color will change when all the food tiles are collected. At the final level the player will turn green indicating the winning status.

**Pull request closed without merging**